### PR TITLE
HYDRA-689 : Add a flow viewport toolkit API .md file

### PR DESCRIPTION
**Pull request by lanierd-adsk**

### CODE REVIEW

**Comment by lanierd-adsk** [comment]
FYI : @lilike-adsk @ppt-adsk @vlasovi @debloip-adsk

File: doc/flowViewportToolkit.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ - Get information about the Hydra viewports and which render delegate they use
+ - Get the version of the API
+ 
+ Each of the domain above is a separate C++ interface with only a few functions in it.
+
+## What is the Flow viewport toolkit merging scene index ?
+
+To be able to do filtering on all data from the DCC native data, USD stages and custom data producer scene indices adding new primitives, we use a merging scene index which gathers all scene indices into a single scene index, please see the following :
+
+![Flow viewport merging scene index schema](https://github.com/Autodesk/maya-hydra/tree/dev/doc/images/fvpMergingSceneIndex.png)
```
Comment (roopavr-adsk) [inline doc/flowViewportToolkit.md]: This link is broken. Maybe a relative path?

File: doc/flowViewportToolkit.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+ - a dccNode is a *MObject** from a DAG node for Maya. If you provide a no null pointer, we automatically track some events from attributes such as transform or visibility updated and apply this change to the primitives from the data producer scene index. If the node gets deleted, we remove the scene index primitives from the merging scene index. If this parameter is a *nullptr*, we won't do anything if the node's attributes changes or the node gets deleted. Basically, this is a way for you to set the DCC node as a parent node for all your primitives from the scene index.
+ - a hydraViewportId which is a Hydra viewport string identifier to which your data producer scene index needs to be associated to. This is a way to add your primitives to only one viewport. Note : set it to *PXR_NS::FvpViewportAPITokens->allViewports* to add this data producer scene index to all viewports. To retrieve a specific hydra viewport identifier, please use the *[InformationInterface](https://github.com/Autodesk/maya-hydra/blob/dev/lib/flowViewport/API/fvpInformationInterface.h)* class.
```
Comment (roopavr-adsk) [inline doc/flowViewportToolkit.md]: Italize/bold this too if you intend to treat is special identifier in the docs?

File: doc/flowViewportToolkit.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+ - a rendererNames which are the Hydra renderer (render delegate)  names to which this scene index should be added. This is only used when hydraViewportId above is set to *PXR_NS::FvpViewportAPITokens->allViewports*, meaning you want to add this scene index to all viewports that are using these renderers. Note : to apply to multiple renderers, use a separator such as : "GL, Arnold". We are actually looking for the render delegate's name in this string. Set this parameter to *PXR_NS::FvpViewportAPITokens->allRenderers* to add your scene index to all viewports whatever their renderer is.
```
Comment (roopavr-adsk) [inline doc/flowViewportToolkit.md:70]: Italize/bold this too if you intend to treat is special identifier in the docs?

**Comment by roopavr-adsk** [comment]
How about converting all the links in the current document to have relative paths?Like we do here:https://github.com/Autodesk/maya-hydra/blob/lanierd/HYDRA-689/doc/selectionHighlightingArchitecture.md

**Comment by lanierd-adsk** [comment]
> How about converting all the links in the current document to have relative paths?Like we do here:https://github.com/Autodesk/maya-hydra/blob/lanierd/HYDRA-689/doc/selectionHighlightingArchitecture.md

What would be the benefit ? For me, it looks safer to have full path in case we move that file to another folder.

**Comment by roopavr-adsk** [comment]
One use case that would benefit is when the repo is being used locally. For ex in VSCode, clicking on link opens the files(say cpp/h) locally in their editor vs opening up the link externally in GitHub.  

Either choices work for me.

**Review by roopavr-adsk — APPROVED** [review]

**Pull request merged**